hpy/devel/abitag.py

```diff
@@ -0,0 +1,63 @@
+import sys
+from distutils import sysconfig
+
+HPY_ABI_TAG = 'hpy0'
+
+def parse_ext_suffix(ext_suffix=None):
+    """
+    Parse EXT_SUFFIX and return abi_tag, ext.
+
+    - abi_tag is a string representing the CPython ABI tag: e.g. 'cp38', 'cp310d', etc.
+    - ext is the filename extension, e.g 'so', 'pyd' or 'dylib'
+    """
+    if ext_suffix is None:
+        ext_suffix = sysconfig.get_config_var('EXT_SUFFIX')
+
+    # ext_suffix is something like this. We want to keep the parts which are
+    # related to the ABI and remove the parts which are related to the
+    # platform:
+    #   - linux:   '.cpython-310-x86_64-linux-gnu.so'
+    #   - mac:     '.cpython-310-darwin.so'
+    #   - win:     '.cp310-win_amd64.pyd'
+    #   - pypy:    '.pypy38-pp73-x86_64-linux-gnu.so'
+    #   - graalpy: '.graalpy-38-native-x86_64-darwin.dylib'
+    assert ext_suffix[0] == '.'
+    _, soabi, ext = ext_suffix.split('.')
+
+    # this is very ad-hoc but couldn't find a better way to do it
+    parts = soabi.split('-')
+    if parts[0].startswith('cpython'):      # cpython on linux, mac
+        n = 2
+    elif parts[0].startswith('cp'):         # cpython on windows
+        n = 1
+    elif parts[0].startswith('pypy'):       # pypy
+        n = 2
+    elif parts[0].startswith('graalpy'):    # graalpy
+        n = 3
+    else:                                   # none of the above, keep all parts
+        n = len(parts)
+
+    abi_tag = '-'.join(parts[:n])
+    # on cpython linux/mac, abi is now cpython-310: shorten it ot cp310, like
+    # on windows and on wheel tags
+    abi_tag = abi_tag.replace('cpython-', 'cp')
```

Comment:
Is it possible to change the value graalpython returns here or has that ship sailed? I think graalpy-38-native is a bit long. This is the name used in the wheel filename, so will wheels be called `numpy-1.23.4-graalpy38-graalpy-38-native-manylinux_2_17_x86_64.whl ` ?